R/R/zzz.R

```diff
@@ -1,22 +1,100 @@
-.onAttach <- function(libname, pkgname) {
-  tryCatch(
-    {
-      metaflow_attach()
-    },
-    error = function(e) {
-      packageStartupMessage("  * Call 'metaflow::install()' to finish installation * ")
-    }
+pkg.env <- new.env()
+
+pkg.env$configs <- list(
+  default = list(
+    metaflow_path = expression(reticulate::py_discover_config("metaflow")$required_module_path)
+  ),
+  batch = list(
+    metaflow_path = expression(path.expand(paste0(getwd(), "/metaflow")))
   )
-  invisible(NULL)
-}
+)
+
+pkg.env$envname = "r-metaflow"
+
+pkg.env$activated = FALSE
 
 .onLoad <- function(libname, pkgname) {
-  tryCatch(
-    {
-      metaflow_load()
-    },
-    error = function(e) {
+  # activate Metaflow conda/virtualenv if they're available
+  # need to call this before check_python_dependencies()
+  pkg.env$activated <- activate_metaflow_env(pkg.env$envname)
+
+  if (pkg.env$activated && check_python_dependencies()) {
+    metaflow_load()
+    print_metaflow_versions()
+  } else {
+    print_metaflow_install_options()
+  }
+}
+
+print_metaflow_install_options <- function(){
+  packageStartupMessage(
+      "* Metaflow Python dependencies not found *\n",
+      "  Available options:\n",
+      "    - Call `install_metaflow()` to install into a new conda or virtualenv\n",
+      "    - Set `METAFLOW_PYTHON` environment variable to the path of your python executable.\n",
+      "      Note: Metaflow needs to be available in the environment specified by `METAFLOW_PYTHON`"
```

Comment:
Not just Metaflow, we need pandas and numpy as well. An alternate messaging could be `- Set 'METAFLOW_PYTHON' env variable to the python executable that has metaflow, numpy, and pandas available as dependencies.`

Reply:
good catch. Will do